lib/Monads/OptionMonadWP.thy

```diff
@@ -299,7 +299,12 @@ lemma no_ofail_wp_comb2[wp_comb]:
   "\<lbrakk> no_ofail P f; no_ofail P' f \<rbrakk> \<Longrightarrow> no_ofail (\<lambda>s. P s \<and> P' s) f"
   by (simp add: no_ofail_def)
 
-
+(* Lemmas relating ovalid and valid *)
+lemma ovalid_gets_the:
+  "ovalid P f Q \<Longrightarrow> \<lbrace>P\<rbrace> gets_the f \<lbrace>Q\<rbrace>"
+  apply wpsimp
+  apply (fastforce dest: use_ovalid)
+  done
```

Comment:
the silly style in this file seems to be two (or more?) lines before a "section" comment and zero after, meaning if we care there should be an extra line before and after your new "section"